### PR TITLE
auto tap validation test for meshed grid with transformers

### PR DESCRIPTION
**Pull request by mgovers**
Add validation test for automatic tap adjustment for a meshed grid with transformers in the mesh

@TonyXiang8787 I added you as a reviewer as you requested

### CODE REVIEW

File: tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json:1]: the input json is the same for all test cases

File: tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      {"id": 9, "u_rated": 400},
+      {"id": 10, "u_rated": 400}
+    ],
+    "line": [
+      {"id": 11, "from_node": 0, "to_node": 1, "from_status": 1, "to_status": 1, "r1": 0.005, "x1": 0.5, "c1": 0, "tan1": 0},
+      {"id": 12, "from_node": 0, "to_node": 2, "from_status": 1, "to_status": 1, "r1": 0.1, "x1": 0.6, "c1": 0, "tan1": 0},
+      {"id": 13, "from_node": 3, "to_node": 5, "from_status": 1, "to_status": 1, "r1": 0.005, "x1": 0.5, "c1": 0, "tan1": 0},
+      {"id": 14, "from_node": 4, "to_node": 6, "from_status": 1, "to_status": 1, "r1": 0.003, "x1": 0.3, "c1": 0, "tan1": 0},
+      {"id": 15, "from_node": 7, "to_node": 9, "from_status": 1, "to_status": 1, "r1": 0.005, "x1": 0.5, "c1": 0, "tan1": 0},
+      {"id": 16, "from_node": 8, "to_node": 9, "from_status": 1, "to_status": 1, "r1": 0.005, "x1": 0.6, "c1": 0, "tan1": 0}
```
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json:26]: i chose to let the lines have various impedances to induce different optimal tap position settings

File: tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/input.json:1]: the grid has the following form:

```
       1 ---oo--- 3 ------ 5 ---oo--- 7
      /                                \
     /                                  \
S - 0                                    9 ---oo--- 10 --- L
     \                                  /
      \                                /
       2 ---oo--- 4 ------ 6 ---oo--- 8
```

**Comment by mgovers** [comment]
docs build fails due to https://github.com/mgaitan/sphinxcontrib-mermaid/pull/161

File: tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/sym_output.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      {"id": 4, "energized": 1, "u_pu": 0.9945461001910171, "u": 9945.461001910171, "u_angle": -2.620069171582145, "p": -100000.000000009, "q": -9999.999999940057},
+      {"id": 5, "energized": 1, "u_pu": 0.9944290227324328, "u": 9944.290227324329, "u_angle": -2.621203754353982, "p": -99999.99999995019, "q": -9999.999999937614},
+      {"id": 6, "energized": 1, "u_pu": 0.9943233999502603, "u": 9943.233999502603, "u_angle": -2.62095218690015, "p": -100000.0000000192, "q": -10000.00000005206},
+      {"id": 7, "energized": 1, "u_pu": 0.9920807600913329, "u": 396.8323040365332, "u_angle": 1.044783740215204, "p": -99999.99999998928, "q": -9999.999999983958},
+      {"id": 8, "energized": 1, "u_pu": 0.992180588914527, "u": 396.8722355658108, "u_angle": 1.044923623034546, "p": -100000.000000007, "q": -10000.00000000539},
+      {"id": 9, "energized": 1, "u_pu": 0.8682948639612919, "u": 347.3179455845167, "u_angle": 0.6381168383951473, "p": -99999.99999999936, "q": -10000.00000000324},
+      {"id": 10, "energized": 1, "u_pu": 0.9636190035727636, "u": 385.4476014291055, "u_angle": -1.979757523221217, "p": -100000.0000000106, "q": -9999.999999998947}
```
Comment (nitbharambe) [inline tests/data/power_flow/automatic-tap-regulator/auto-tap-changer-meshed-any/sym_output.json:18]: To keep the motivation of tests concise, can you remove the powers?

**Review by nitbharambe — APPROVED** [review]
Lets keep in mind to think of reusing input files for different outputs in future becomes a good idea. We now have many duplicate input files in all automatic tap changer cases.

**Comment by mgovers** [comment]
@TonyXiang8787 awaiting your approval (no hurry)

**Pull request merged**